File: Interviews/GFG/Daily-Problems/2022/November/Day-23/Maximum-Sum-LCM.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
   public:
    long long int maxSumLCM(int n) {
        long long int ans = 0;
        for (int i = 1; i * i <= n; i++) {
            if (n % i == 0) {
                ans += i;
                if (n / i != i) {
                    ans += n / i;
                }
            }
        }

        return ans;
    }
};

File: Interviews/GFG/Daily-Problems/2022/November/Day-23/Maximum-Sum-LCM.py
class Solution:
    def maxSumLCM(self, n):
        ans, i = 0, 1
        while i * i <= n:
            if n % i == 0:
                ans += i
                if n // i != i:
                    ans += n // i
            i += 1

        return ans
